File: code/design-patterns/behavioral/observer/python/observer_usage.py
from observer import Store, Customer


store = Store()

customer1 = Customer("Customer 1")
customer2 = Customer("Customer 2")

store.add_subscriber(customer1)
store.add_subscriber(customer2)

store.notify()
# Outputs:
# Customer 1 has been notified.
# Customer 2 has been notified.

store.remove_subscriber(customer1)

store.notify()
# Outputs:
# Customer 2 has been notified.

File: code/design-patterns/behavioral/observer/go/observer_usage.go
package main

func main() {
	store := &Store{}

	customer1 := &Customer{name: "Customer 1"}
	customer2 := &Customer{name: "Customer 2"}

	store.addSubscriber(customer1)
	store.addSubscriber(customer2)

	store.notify()
	// Outputs:
	// Customer 1 has been notified.
	// Customer 2 has been notified.

	store.removeSubscriber(customer1)

	store.notify()
	// Outputs:
	// Customer 2 has been notified.
}
